go/implementations/mock_record_reader.go

```diff
@@ -0,0 +1,39 @@
+package implementations
+
+import (
+	"github.com/apache/arrow-go/v18/arrow"
+)
+
+// MockRecordReader for testing purposes
+type MockRecordReader struct {
```

Comment:
Do you have an interface, which this structure should implement? In golang it's a good practice to validate, that interface is correctly implemented in compilation level. 
like this:
`var _ Somether = (*MyType)(nil)`

Could you please add this to all the services you are implementing interfaces in? This is a compile time check, which will be handy